test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,212 @@
+            return False
+        return True
+
+    def amqp_callback(self, ch, method, properties, body):
+        logs.debug_3("Routing Key %r:" % method.routing_key)
+        logs.data_log.debug_3(body.__str__())
+        global amqp_message_received
+        global routing_key, amqp_body
+        amqp_message_received = True
+        amqp_body = body
+        routing_key = method.routing_key
+
+    def test_create_racknode_amqp_messages(self):
+        logs.debug_3("Going to generate 10 rack nodes to trigger AMQP message")
+        for operator in range(0, 10):
+            rackname = "myRack" + "_" + datetime.now().__str__()
+            newrack = {"name": rackname, "type": "rack"}
+            logs.debug_3(" Create New Tag %r" % newrack)
+            mon_url = '/api/2.0/nodes'
+            # start amqp thread
+            global amqp_message_received
+            amqp_message_received = False
+            logs.debug_3('launch AMQP thread')
+            td = AmqpWorker(
+                exchange_name="on.events", topic_routing_key="node.added.#",
+                external_callback=self.amqp_callback, timeout=10)
+            td.setDaemon(True)
+            td.start()
+            mon_data_post = fit_common.rackhdapi(mon_url, action='post', payload=newrack)
+            self.assertIn(
+                mon_data_post['status'], range(200, 205),
+                "Incorrect HTTP return code: {}".format(mon_data_post['status']))
+            rackid = mon_data_post['json']['id']
+            mon_url = '/api/2.0/nodes/{}'.format(rackid)
+            mon_data = fit_common.rackhdapi(mon_url)
+            self.assertIn(
+                mon_data['status'], range(200, 205),
+                "Incorrect HTTP return code: {}".format(mon_data['status']))
+            json_node_data = mon_data['json']
+            self.assertTrue(
+                json_node_data['name'] == newrack['name'] and json_node_data['type'] == "rack",
+                "rack node field error")
+            timecount = 0
+            while amqp_message_received is False and timecount < 10:
+                sleep(1)
+                timecount = timecount + 1
+            self.assertNotEquals(timecount, 10, "No AMQP message received")
+            expected_key = "node.added.information." + rackid + '.' + rackid
+            expected_payload = {
+                "type": "node",
+                "action": "added",
+                "typeId": rackid,
+                "nodeId": rackid,
+                "severity": "information",
+                "version": "1.0",
+                "createdAt": mon_data_post['json']['createdAt']}
+            self.assertEquals(self._compare_message(expected_key, expected_payload), True, "AMQP Message Check Error!")
+            logs.debug_3("query rack: %r successfully!" % rackname)
+        logs.debug_2("test: rack creation and query succeed!")
+
+    def test_delete_rack_amqp_messages(self):

```

Comment:
What if there is no node found? Strongly suggest using depend on method, and run this case after the first one finished.